elastic/beats
# [Metricbeat/Kibana/stats] `exclude_usage=true` by default

## What does this PR do?


It adds changes to the Metricbeat's `kibana` module: it adds a new configuration flag `stats.exclude_usage` (default `true`) that is used in the metricset `stats` in the method `shouldCollectUsage` to decide whether it should collect usage (no matter the time).

The reason for this change is to disable the usage collection via the `/api/stats` endpoint but to still allow the collection of the usage via explicitly setting the flag to `false` if necessary.

## Why is it important?


We are working towards decoupling Monitoring and Telemetry in Kibana. At the moment, Monitoring holds in the `.monitoring-kibana-*` indices the Kibana's Usage details only to be able to report them when sending the telemetry reports. This data is already sent to the remote telemetry cluster via Kibana's `local` collectors, so there's no need for this duplication in the efforts. Thanks to these changes, Monitoring will only maintain in its indices the data they really need for its own features.

## Checklist




## Author's Checklist


- [ ] Did I filled the default config in the right way?
- [ ] Not a Go developer myself. Did I mess anything up? 😇 

## How to test this PR locally


Setup Metricbeat with the `kibana-xpack` module enabled and notice it never asks for the `/api/stats?extended&legacy&exclude_usage=false` request (the last query is always `true` by default).

## Related issues


- Closes #22651
- Relates https://github.com/elastic/kibana/issues/83521
- Relates https://github.com/elastic/kibana/issues/83626

## Use cases


Explained in the issue #22651

## Screenshots



## Logs
